src/rpc/iterator/ExtGenFilm.h

```diff
@@ -0,0 +1,157 @@
+#ifndef RPC_EXT_GEN_FILM_H
+#define RPC_EXT_GEN_FILM_H
+
+/*
+* PSCF - Polymer Self-Consistent Field Theory
+*
+* Copyright 2016 - 2022, The Regents of the University of Minnesota
+* Distributed under the terms of the GNU General Public License.
+*/
+
+#include <prdc/iterator/ExtGenFilmBase.h>  // Base class
+#include <rpc/System.h>
+
+namespace Pscf {
+namespace Rpc {
+
+   using namespace Util;
+   using namespace Pscf::Prdc;
+
+   /**
+   * External field generator for a thin film geometry.
+   *
+   * The parent ExtGenFilmBase class template defines all traits of a 
+   * ExtGenFilm that do not require access to the System. This subclass
+   * defines all methods that need System access.
+   * 
+   * If the user chooses an ExtGenFilm object to generate external fields,
+   * the external fields will have the same shape as the mask, with a 
+   * magnitude defined by a Flory--Huggins-like chi parameter. This class
+   * is specific to thin-film systems because it also allows for a 
+   * different chi parameter to be defined on the top boundary than on
+   * the bottom, through user input arrays chi_bottom and chi_top. See 
+   * \ref user_thin_films_page for more information. 
+   *
+   * \ingroup Rpc_Iterator_Module
+   */
+   template <int D>
+   class ExtGenFilm : public ExtGenFilmBase<D>
+   {
+
+   public:
+
+      /**
+      * Default constructor
+      */
+      ExtGenFilm();
+      
+      /**
+      * Constructor
+      * 
+      * \param itr  Iterator parent object
+      */
+      ExtGenFilm(System<D>& sys);
+
+      /**
+      * Destructor
+      */
+      ~ExtGenFilm();
+
+      /**
+      * Check whether the fields have been generated
+      */
+      bool isGenerated() const;
+
+      using ExtGenFilmBase<D>::isAthermal;
+      using ExtGenFilmBase<D>::chiBottom;
+      using ExtGenFilmBase<D>::chiTop;
+      using ParamComposite::setClassName;
+
+   protected:
+
+      /**
+      * Allocate container necessary to generate and store fields
+      */ 
+      void allocate();
+
+      /**
+      * Generate the fields and store where the Iterator can access.
+      */
+      void generate();
+
+      /**
+      * Get the System associated with this object by reference.
+      */
+      System<D>& system();
+      /**
+      * Get the System associated with this object by const reference.
+      */
+      System<D> const & system() const;
+
+      /**
+      * Get the space group name for this system.
+      */
+      std::string getSpaceGroup() const;
```

Comment:
The use of "get" in the names of these and a few other related functions seems confusing. It is common in some styles to pre-pend the word "get" to any accessor that simply returns the value of a class member variable. PSCF doesn't use this style, and instead generally uses the variable name (with no "get") as the name of a function that just returns a private class member by value or reference. It appears that here, these functions are retrieving values from the associated System, but it took me a while to figure out that this appears to be why you used a "get" prefix. Please consider two possibilities:

      A.  If these functions are only used in a few places in this class and its subclasses, maybe it would be clearer to not define this wrapper function and just call the system().domain(). .... as needed.
      B. If you think it is important to have these wrapper function, consider replacing "get" by "system", calling it "systemSpaceGroup" for example.

My guess is that A (no wrapper) may end up being easier to read, but it's up to you.